src/app/modules/group/http-services/remove-group-manager.service.ts

```diff
@@ -14,4 +25,22 @@ export class RemoveGroupManagerService {
   remove(groupId: string, managerId: string): Observable<SimpleActionResponse> {
     return this.http.delete<SimpleActionResponse>(`${appConfig.apiUrl}/groups/${groupId}/managers/${managerId}`);
   }
+
+  removeBatch(parentGroupId: string, ids: string[], ownManagerId?: string): Observable<Result> {
+    return merge(...ids.map(id => this.remove(parentGroupId, id))).pipe(
+      catchError(() => of(failedState)),
```

Comment:
Here, when a request fails, all the pending ones are cancelled because an error is thrown at the global (merged) observable level. That error causes the whole observable to stop, which cancels treatment for other pending requests

I think the catchError should be applied to each "remove" observable (to be tested):
```ts
return merge(...ids.map(id => this.remove(parentGroupId, id).pipe(catchError(…)).pipe(
  reduce(…)
)
```
PS: I know I told you recently to keep observables as flat as possible, but sometimes it's not possible :confused: